### PR TITLE
Fix ST_NumPoints function for geometry collections

### LINKED ISSUE #10625: ST_NumPoints fails with NullPointerException for geometry collections
```
presto> select ST_NumPoints(ST_GeometryFromText('GEOMETRYCOLLECTION (POINT (1 2), LINESTRING (8 4, 5 7), POLYGON EMPTY)'));
Query 20180517_012104_06300_p8fjg failed: null
java.lang.NullPointerException
        at com.facebook.presto.plugin.geospatial.GeoFunctions.stNumPoints(GeoFunctions.java:390)
```

### PR DESCRIPTION
**Pull request by mbasmanova**
Fix NPE in the ST_NumPoints function when input is a geometry collection.

Fixes #10625

### CODE REVIEW

File: presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java [hunk 1]
```diff
@@ -404,13 +404,26 @@ public void testSTNumInteriorRing()
...
+        assertNumPoints("LINESTRING (8 4, 5 7)", 2);
+        assertNumPoints("MULTILINESTRING ((1 1, 5 1), (2 4, 4 4))", 4);
+        assertNumPoints("POLYGON ((0 0, 8 0, 0 8, 0 0), (1 1, 1 5, 5 1, 1 1))", 6);
+        assertNumPoints("MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1)), ((2 4, 2 6, 6 6, 6 4)))", 8);
+        assertNumPoints("GEOMETRYCOLLECTION (POINT (1 2), LINESTRING (8 4, 5 7), POLYGON EMPTY)", 3);
+    }
+
+    private void assertNumPoints(String wkt, int expectedPoints)
+    {
+        assertFunction("ST_NumPoints(ST_GeometryFromText('" + wkt + "'))", BIGINT, (long) expectedPoints);
```
Comment (arhimondr) [inline presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java]: use `format` here

File: presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java [hunk 2]
```diff
@@ -507,14 +507,25 @@ public static Slice stGeometryN(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlTy
     @SqlType(StandardTypes.BIGINT)
     public static long stNumPoints(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        OGCGeometry geometry = deserialize(input);
-        if (geometry.getEsriGeometry().isEmpty()) {
-            return 0;
-        }
-        else if (GeometryType.getForEsriGeometryType(geometry.geometryType()) == GeometryType.POINT) {
-            return 1;
+        GeometryCursor cursor = deserialize(input).getEsriGeometryCursor();
```
Comment (arhimondr) [inline presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java]: You can use an utility method introduced in #10626

**Comment by mbasmanova** [comment]
@arhimondr Addressed comments. Andrii, would you take another look?

File: presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java [hunk 3]
```diff
@@ -507,14 +508,7 @@ public static Slice stGeometryN(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlTy
...
     {
-        OGCGeometry geometry = deserialize(input);
-        if (geometry.getEsriGeometry().isEmpty()) {
-            return 0;
-        }
-        else if (GeometryType.getForEsriGeometryType(geometry.geometryType()) == GeometryType.POINT) {
-            return 1;
-        }
-        return ((MultiVertexGeometry) geometry.getEsriGeometry()).getPointCount();
+        return GeometryUtils.getPointCount(deserialize(input));
```
Comment (arhimondr) [inline presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java]: static import

File: presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestBingTileFunctions.java [hunk 4]
```diff
@@ -319,6 +320,11 @@ public void testGeometryToBingTiles()
 
         assertFunction("transform(geometry_to_bing_tiles(ST_Envelope(ST_GeometryFromText('LINESTRING (59.765625 29.84064389983442, 60.2 30.14512718337612)')), 10), x -> bing_tile_quadkey(x))", new ArrayType(VARCHAR), ImmutableList.of("1230301230", "1230301231"));
 
+        // Empty geometries
+        assertGeometryToBingTiles("POINT EMPTY", 10, emptyList());
+        assertGeometryToBingTiles("POLYGON EMPTY", 10, emptyList());
+        assertGeometryToBingTiles("GEOMETRYCOLLECTION EMPTY", 10, emptyList());
```
Comment (arhimondr) [inline presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestBingTileFunctions.java]: Add a tests for geometry collection that contains empty geometry. Sorry, totally missed that during the first pass.

**Pull request merged**